x-pack/plugins/maps/public/elasticsearch_geo_utils.js

```diff
@@ -225,41 +225,80 @@ export function geoShapeToGeometry(value, accumulator) {
   accumulator.push(geoJson);
 }
 
-function createGeoBoundBoxFilter(geometry, geoFieldName, filterProps = {}) {
-  ensureGeometryType(geometry.type, [GEO_JSON_TYPE.POLYGON]);
+function createGeoBoundBoxFilter({ maxLat, maxLon, minLat, minLon }, geoFieldName) {
```

Comment:
consider not actually splitting. ES uses this specification for bounding boxes http://docs.opengeospatial.org/is/12-063r5/12-063r5.html#30

> However when the area crosses the 180° meridian, the value of the lower left longitude will be greater than the value of the upper right longitude

This is a real elaborate specification for such a simple thing, sinde an east->west bounding box - e.g. `[-220,40,-170,30]` like in geojson, mapbox-gl, leaflet ... are unambiguous how they need to be interpreted, with the added advantage that they work in a cartesian system.

But I think we could simplify to:


```
  let esBbox;
  if (maxLon - minLon >= 360) {
    console.log('span blove');
    esBbox = {
      top_left: [-180, top],
      bottom_right: [180, bottom],
    });
  } else {
    console.log('does not span globe');
    //does not span globe
    const newMinlon = ((minLon + 180 + 360) % 360) - 180;
    const newMaxlon = ((maxLon + 180 + 360) % 360) - 180;
    console.log(newMinlon < newMaxlon ? 'does not span dateline' : 'spans dateline');
    esBbox = {
      top_left: [newMinlon, top],
      bottom_right: [newMaxlon, bottom],
    });
  }
  return {
    geo_bounding_box: {
      [geoFieldName]: esBbox,
    },
  };
```

It would also be more in line with ES-bbox handling here https://github.com/elastic/kibana/blob/1252b832f45ca02af17543040e2ea2be0dc6fecf/x-pack/plugins/maps/public/layers/sources/es_source/es_source.js#L180-L185

Reply:
Thanks for the clarification. This is actually the old logic. This PR really just removes the logic when building geo_shape queries which do not need this type of work